src/storage/sqlite/xerial/src/main/java/org/locationtech/geogig/storage/sqlite/XerialObjectDatabaseV2.java

```diff
@@ -352,10 +353,12 @@ public void putAll(Iterator<? extends RevObject> objects, final BulkOpListener l
         checkWritable();
 
         // System.err.println("put allllllll");
-        txHandler.startTransaction();
-        while (objects.hasNext()) {
+        //txHandler.startTransaction();
```

Comment:
@groldan The change to UnmodifiableIterator seems to address the other concurrency issues I was seeing with the DataStore tests that caused me to comment out the _startTransaction()_ call here. I'm going to try to revert some of this as well and see if the tests pass now. Re-enabling this should give us better performance for when batch edits happen in large chunks.